### PR TITLE
Pin Bicep to v0.4.1272

### PR DESCRIPTION
**Pull request by glennmusa**
# Description

This change:

- Updates the devcontainer Dockerfile to accept a `BICEP_VERSION` argument and pins it to v0.4.1272
- Updates the `validate-build-bicep` workflow to set a `BICEP_VERSION` argument and pins it to v0.4.1272
- Makes updates to source to fix `explicit-values-for-loc-params` Bicep warnings that occur when upgrading Bicep to v0.4.1272

## Issue reference

The issue this PR will close: #646 

## Checklist

Please make sure you've completed the relevant tasks for this PR out of the following list:

* [x] All acceptance criteria in the backlog item are met
* [x] The documentation is updated to cover any new or changed features
* [x] Manual tests have passed
* [x] Relevant issues are linked to this PR

### CODE REVIEW

File: src/bicep/modules/hubNetwork.bicep [hunk 1]
```diff
@@ -240,6 +240,7 @@ module azureMonitorPrivateLink './privateLink.bicep' = if ( contains(supportedCl
     logAnalyticsWorkspaceResourceId: logAnalyticsWorkspaceResourceId
     privateEndpointSubnetName: subnetName
     privateEndpointVnetName: virtualNetwork.outputs.name
+    location: location
```
Comment (brooke-hamilton) [inline src/bicep/modules/hubNetwork.bicep:243]: This PR contains some changes related to how location is set on policy and networking. They do not seem related to the purpose of the PR. Did you intend to include these changes.? They appear to make sense, but maybe they are part of another effort?

**Review by brooke-hamilton — CHANGES_REQUESTED** [review]

File: src/bicep/modules/hubNetwork.bicep (hunk 1 above)
Comment (glennmusa) [reply src/bicep/modules/hubNetwork.bicep:243]: Good catch! Yes, these changes are intentional.

These changes are the cost of updating to Bicep v0.4.1272 which introduces new warnings for implied resource locations.

> Makes updates to source to fix explicit-values-for-loc-params Bicep warnings that occur when upgrading Bicep to v0.4.1272

You can see what happens without these changes in this failed `validate-build-bicep` run here:

https://github.com/Azure/missionlz/runs/5208013963?check_suite_focus=true#step:3:19

Comment (brooke-hamilton) [reply src/bicep/modules/hubNetwork.bicep:243]: 👍 that makes sense. It looks great. 😻 I'm running some test deployments.

**Review by brooke-hamilton — APPROVED** [review]
✨
